src/vm/analysis/read_only_checker/mod.rs

```diff
@@ -290,27 +341,31 @@ impl<'a, 'b> ReadOnlyChecker<'a, 'b> {
                     _ => return Err(CheckError::new(CheckErrors::ContractCallExpectName)),
                 };
 
-                self.check_all_read_only(&args[2..])
+                self.check_each_expression_is_read_only(&args[2..])
                     .map(|args_read_only| args_read_only && is_function_read_only)
             }
         }
     }
 
-    fn check_function_application_read_only(
+    /// Checks the native function application implied by `expressions`. The first
+    /// argument is used as the function name, and the tail is used as the arguments.
+    ///
+    /// Returns `true` iff the function application is read-only.
```

Comment:
Could add section for Errors here & mention the possible `Unknown Function` error.

Reply:
thanks!